cfme/tests/services/test_dialog_regex_validation_in_catalog.py

```diff
@@ -169,4 +169,65 @@ def test_regex_dialog_validation_error():
             5.
             6. It shouldn't gives the validation error
     """
-    pass
+    service_dialog = appliance.collections.service_dialogs
+    element_data = {
+        "element_information": {
+            "ele_label": fauxfactory.gen_alphanumeric(15, start="ele_label_"),
+            "ele_name": fauxfactory.gen_alphanumeric(15, start="ele_name_"),
+            "ele_desc": fauxfactory.gen_alphanumeric(15, start="ele_desc_"),
+            "choose_type": "Text Box",
+        },
+        "options": {"validation_switch": True, "validation": "^[0-9]*$"},
+    }
+    sd = service_dialog.create(
+        label=fauxfactory.gen_alphanumeric(start="dialog_"), description="my dialog"
+    )
+    tab = sd.tabs.create(
+        tab_label=fauxfactory.gen_alphanumeric(start="tab_"), tab_desc="my tab desc"
+    )
+    box = tab.boxes.create(
+        box_label=fauxfactory.gen_alphanumeric(start="box_"), box_desc="my box desc"
+    )
+    box.elements.create(element_data=[element_data])
+
+    navigate_to(sd, "Edit")
+    view = appliance.browser.create_view(EditElementView)
+    label = element_data["element_information"]["ele_label"]
+    view.element.edit_element(label)
+    view.options.click()
+    assert view.options.validation_switch.fill(False)
+    view.ele_save_button.click()
+    view.save_button.click()
+
+    catalog_item = appliance.collections.catalog_items.create(
+        appliance.collections.catalog_items.GENERIC,
+        name=fauxfactory.gen_alphanumeric(15, start="cat_item_"),
+        description="my catalog",
+        display_in=True,
+        catalog=catalog,
+        dialog=sd)
+
+    @request.addfinalizer
+    def _cleanup():
+        if catalog_item.exists:
+            catalog_item.delete()
+        sd.delete_if_exists()
+
+    ele_name = element_data["element_information"]["ele_name"]
+    service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog, catalog_item.name)
+    view = navigate_to(service_catalogs, 'Order')
+
+    input_data_list = [fauxfactory.gen_alpha(length=3),
+                       fauxfactory.gen_number(),
+                       fauxfactory.gen_special(length=5),
+                       fauxfactory.gen_special(length=5)
+                       ]
+
+    msg = f"""Entered text should match the format: {element_data["options"]["validation"]}"""
+
+    for input_data in input_data_list:
+        logger.info('Entering input data: %s ' % input_data)
+        view.fields(ele_name).fill(input_data)
+        element = view.fields(ele_name).input
+        assert element.warning != msg
```

Comment:
Still I won't found condtion where we can match exact warning i.e. input not satisfy provided regex exp.

is it not possible to add one assertion which will raise warning and we can assert msg?
`assert element.warning == msg`

Reply:
We can not assert the exact message/input based message in the assertion because for every input we get the same warning if any. added image where we can get a warning, I collected this image from another scenario, this is not related to our test case scenario. It is just for a better understanding  
![image](https://user-images.githubusercontent.com/23401179/80242834-a37df480-8683-11ea-92bf-21d2bb763596.png)




In this test case we can not raise any warning because we disabled regex pattern in the element.

Reply:
What problem we are facing? Its looks easy to add this assertion. `Entered text should match the format: {regex}`?

Reply:
As per IRC discussion. test is with disabling validation. resolving thread.
Need better test name.